dev/com.ibm.ws.microprofile.config/src/com/ibm/ws/microprofile/config/converters/DefaultConverters.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+//This class was inspired by com.netflix.archaius.DefaultDecoder
+
+package com.ibm.ws.microprofile.config.converters;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.Period;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ServiceConfigurationError;
+import java.util.ServiceLoader;
+
+import org.eclipse.microprofile.config.spi.Converter;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.microprofile.config.interfaces.ConfigException;
+
+/**
+ * The helper class returns all the built-in converters.
+ *
+ */
+public class DefaultConverters {
+
+    private static final TraceComponent tc = Tr.register(DefaultConverters.class);
+
+    private static final PriorityConverterMap defaultConverters = new PriorityConverterMap();
+
+    static {
+        defaultConverters.addConverter(new IdentityConverter()); // v -> v
+
+        defaultConverters.addConverter(new BooleanConverter());
```

Comment:
With the current Config 1.2 implicit converter support, basically there is no need to add the converter implicitly. If no converter is found, it should automatically fallback to ctor, valueOf, parse. In this way, the code is much cleaner.